daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -2066,4 +2362,47 @@ private[lf] object DecodeV1 {
       .toMap
       .withDefault(_ => throw Error.Parsing("BuiltinFunction.UNRECOGNIZED"))
 
+  // stack-safety achieved via a Work trampoline.
+  private[archive] sealed abstract class Work[A]
```

Comment:
```suggestion
  private sealed abstract class Work[A]
```

Reply:
Yeah. I also wanted this... but I knew `scalac` objected.

But I fought a little bit harder.. and found out why... and it turns out there was one more place where the use of a function returning `Work` had leaked and was being called form test code... and that was wrong.

So fixed now, and `Work` is fully private.

Thanks for making me consider this again!